src/main/java/com/puppycrawl/tools/checkstyle/checks/AvoidEscapedUnicodeCharactersCheck.java

```diff
@@ -179,15 +179,7 @@
 
     /** Regular expression for all escaped chars. */
     private static final Pattern ALL_ESCAPED_CHARS = Pattern.compile("^((\\\\u)[a-fA-F0-9]{4}"
-            + "|\""
-            + "|'"
-            + "|\\\\"
-            + "|\\\\b"
-            + "|\\\\f"
-            + "|\\\\n"
-            + "|\\\\r"
-            + "|\\\\t"
-            + ")+$");
+            + "|[\"'\\\\\\\\b\\\\f\\\\n\\\\r\\\\t])+$");
```

Comment:
Isn't `\\\\b` mean we are looking for the text string `\{binary character}`?
Won't putting this directly into `[]` mean we change search from the above to looking for the text string `\` or the binary character and not the combination of the 2?

Reply:
We should have a test case to cover this.

Reply:
I do not understand what is `\b` at all .

it was added at https://github.com/checkstyle/checkstyle/commit/39f40e57858c8a5c6310bf35e6649b03188b6996 without any specific reason.

Reply:
ok it is `\b | Insert a backspace in the text at this point.`

https://docs.oracle.com/javase/tutorial/java/data/characters.html

Reply:
@rnveach , I do not see problem here 
```
$ javac Test.java 
$ cat Test.java 
public class Test {
  String value = "\u03bc\b"; // ok
}

$ cat config.xml 
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
          "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
          "http://www.puppycrawl.com/dtds/configuration_1_3.dtd">

<module name="Checker">
    <property name="charset" value="UTF-8"/>

    <module name="TreeWalker">
      <module name="AvoidEscapedUnicodeCharacters">
        <property name="allowIfAllCharactersEscaped" value="true"/>
      </module>

    </module>
</module>

$ java -jar checkstyle-8.38-all.jar -c config.xml Test.java
Starting audit...
Audit done.

$ cat config.xml 
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
          "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
          "http://www.puppycrawl.com/dtds/configuration_1_3.dtd">

<module name="Checker">
    <property name="charset" value="UTF-8"/>

    <module name="TreeWalker">
      <module name="AvoidEscapedUnicodeCharacters">
      </module>

    </module>
</module>

$ java -jar checkstyle-8.38-all.jar -c config.xml Test.java
Starting audit...
[ERROR] /var/tmp/Test.java:2:18: Unicode escape(s) usage should be avoided. [AvoidEscapedUnicodeCharacters]
Audit done.
Checkstyle ends with 1 errors.

```

Reply:
Master:
````
$ cat TestClass.java
public class Test {
  String value = "b\u03bc"; // ok
}

$ cat TestConfig.xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
          "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
          "https://checkstyle.org/dtds/configuration_1_3.dtd">

<module name="Checker">
    <property name="charset" value="UTF-8"/>

    <module name="TreeWalker">
      <module name="AvoidEscapedUnicodeCharacters">
<property name="allowIfAllCharactersEscaped" value="true" />
      </module>

    </module>
</module>

$ java -jar checkstyle-8.38-all.jar -c TestConfig.xml TestClass.java
Starting audit...
[ERROR] TestClass.java:2:18: Unicode escape(s) usage should be avoided. [AvoidEscapedUnicodeCharacters]
Audit done.
Checkstyle ends with 1 errors.
````


PR:
````
$ cat TestClass.java
public class Test {
  String value = "b\u03bc"; // ok
}

$ cat TestConfig.xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
          "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
          "https://checkstyle.org/dtds/configuration_1_3.dtd">

<module name="Checker">
    <property name="charset" value="UTF-8"/>

    <module name="TreeWalker">
      <module name="AvoidEscapedUnicodeCharacters">
<property name="allowIfAllCharactersEscaped" value="true" />
      </module>

    </module>
</module>

$ java -jar checkstyle-checkstyle-pr-sonarregexpstackoverflow2-2020-12-26-all.jar -c TestConfig.xml TestClass.java
Starting audit...
Audit done.
````

Short Answer: You can't use `[]`s here.

Reply:
thanks a lot!
UTs are extended to cover this case.